tests/integration/pkg/proxy/proxy.go

```
@@ -0,0 +1,137 @@
+		http.Error(w, "only CONNECT is supported", http.StatusMethodNotAllowed)
+		return
+	}
+
+	p.recordHost(r.Host)
+
+	destConn, err := net.DialTimeout("tcp", r.Host, 10*time.Second)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadGateway)
+		return
+	}
+	defer destConn.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `destConn.Close` is not checked

(errcheck)